media/whip_server.go

```diff
@@ -65,7 +64,7 @@ type WHIPServer struct {
 }
 
 // handleCreate implements the POST that creates a new resource.
-func (s *WHIPServer) CreateWHIP(ctx context.Context, ssr *SwitchableSegmentReader, whepURL string, w http.ResponseWriter, r *http.Request) *MediaState {
+func (s *WHIPServer) CreateWHIP(ctx context.Context, ssr *SwitchableSegmentReader, whepURL string, w http.ResponseWriter, r *http.Request, statsReceiver func(stats *MediaStats)) *MediaState {
```

Comment:
Instead of yet another parameter, for a deeply nested callback, which spins off yet another goroutine - maybe we can just add a `WhipConnection.Stats()` method (which would proxy `MediaState.Stats()`), and hoist the 5s timing loop out of the WhipServer and into the top level handler. Then we can do all the other processing in there in a straight line - write to prometheus, logs, kafka, etc.